source/fab/util.py

```diff
@@ -127,18 +147,67 @@ class CompiledFile(object):
     A Fortran or C file which has been compiled.
 
     """
-    def __init__(self, analysed_file, output_fpath: Path):
+    def __init__(self, input_fpath, output_fpath,
+                 source_hash=None, flags_hash=None, module_deps_hashes: Dict[str, int] = None):
         """
-        :param analysed_file:
+        :param input_fpath:
             The file that was compiled.
         :param output_fpath:
-            The object file.
+            The object file that was created.
+
+        """
+        # todo: Should just be the input_fpath, not the whole analysed file
+        self.input_fpath = Path(input_fpath)
+        self.output_fpath = Path(output_fpath)
```

Comment:
Is there a reason why these are not just passed as `Path` objects?

Reply:
Future robustness/flexibility in case a user-defined step starts adding CompiledFile objects and doesn't pass in paths. Always happy to reconsider.